### PR TITLE
[FIX] Reapply some changes that got lost due to seqan/seqan3#1914

### PR DESCRIPTION
**Pull request by marehr**
It seems that https://github.com/seqan/seqan3/pull/1914 unfortunately didn't start from a fresh checkout of `upstream/master`. It seems that some of my previous changes / fixes got lost due to merge-conflicts.

This PR is re-applying fix

* https://github.com/seqan/seqan3/pull/1900 and
* https://github.com/seqan/seqan3/pull/1885.

Furthermore, I did some code clean-up.

### CODE REVIEW

File: include/seqan3/range/views/pairwise_combine.hpp [hunk 1]
```diff
@@ -65,18 +64,12 @@ class pairwise_combine_view : public std::ranges::view_interface<pairwise_combin
...
-    //!\brief Default Move-Assignment.
-    constexpr pairwise_combine_view & operator=(pairwise_combine_view &&) = default;
-    //!\brief Default Destructor.
-    ~pairwise_combine_view() = default;
+    pairwise_combine_view() = default; //!< Defaulted.
+    pairwise_combine_view(pairwise_combine_view const &) = default; //!< Defaulted.
+    pairwise_combine_view(pairwise_combine_view &&) = default; //!< Defaulted.
+    pairwise_combine_view & operator=(pairwise_combine_view const &) = default; //!< Defaulted.
+    pairwise_combine_view & operator=(pairwise_combine_view &&) = default; //!< Defaulted.
+    ~pairwise_combine_view() = default; //!< Defaulted.
```
Comment (MitraDarja) [inline include/seqan3/range/views/pairwise_combine.hpp:72]: Why are these not constexpr anymore?

File: include/seqan3/range/views/repeat.hpp [hunk 2]
```diff
@@ -88,12 +88,12 @@ class repeat_view : public std::ranges::view_interface<repeat_view<value_t>>
...
-    constexpr repeat_view(repeat_view const &) = default;             //!< Defaulted.
-    constexpr repeat_view & operator=(repeat_view const &) = default; //!< Defaulted.
-    constexpr repeat_view(repeat_view &&) = default;                  //!< Defaulted.
-    constexpr repeat_view & operator=(repeat_view &&) = default;      //!< Defaulted.
-    ~repeat_view() = default;                                         //!< Defaulted.
+    repeat_view() = default; //!< Defaulted.
+    repeat_view(repeat_view const &) = default; //!< Defaulted.
+    repeat_view & operator=(repeat_view const &) = default; //!< Defaulted.
+    repeat_view(repeat_view &&) = default; //!< Defaulted.
+    repeat_view & operator=(repeat_view &&) = default; //!< Defaulted.
```
Comment (MitraDarja) [inline include/seqan3/range/views/repeat.hpp:95]: Again, why are they not constexpr anymore?

File: include/seqan3/range/views/repeat.hpp [hunk 3]
```diff
@@ -205,14 +205,14 @@ class repeat_view : public std::ranges::view_interface<repeat_view<value_t>>
 
 private:
     //!\brief A std::views::single over the input.
-    decltype(std::views::single(std::declval<value_t &>())) single_value;
-}; //class repeat_view
```
Comment (MitraDarja) [inline include/seqan3/range/views/repeat.hpp:209]: Why did you delete this comment? Isn't it helpful to now what this bracket closes?

File: include/seqan3/range/views/repeat.hpp [hunk 4]
```diff
@@ -294,7 +294,7 @@ class repeat_view<value_t>::repeat_view_iterator : public detail::random_access_
         return true;
     }
     //!\}
-}; // class repeat_view::repeat_view_iterator
```
Comment (MitraDarja) [inline include/seqan3/range/views/repeat.hpp:297]: Again, why did you delete this comment?

File: include/seqan3/range/views/repeat.hpp [hunk 5]
```diff
@@ -235,12 +235,12 @@ class repeat_view<value_t>::repeat_view_iterator : public detail::random_access_
...
-    constexpr repeat_view_iterator(repeat_view_iterator const &) = default;             //!< Defaulted.
-    constexpr repeat_view_iterator & operator=(repeat_view_iterator const &) = default; //!< Defaulted.
-    constexpr repeat_view_iterator (repeat_view_iterator &&) = default;                 //!< Defaulted.
-    constexpr repeat_view_iterator & operator=(repeat_view_iterator &&) = default;      //!< Defaulted.
-    ~repeat_view_iterator() = default;                                                  //!< Defaulted.
+    repeat_view_iterator() = default; //!< Defaulted.
+    repeat_view_iterator(repeat_view_iterator const &) = default; //!< Defaulted.
+    repeat_view_iterator & operator=(repeat_view_iterator const &) = default; //!< Defaulted.
+    repeat_view_iterator (repeat_view_iterator &&) = default; //!< Defaulted.
+    repeat_view_iterator & operator=(repeat_view_iterator &&) = default; //!< Defaulted.
```
Comment (MitraDarja) [inline include/seqan3/range/views/repeat.hpp:242]: Why is constexpr deleted?

File: include/seqan3/range/views/take.hpp [hunk 6]
```diff
@@ -238,22 +238,27 @@ class view_take : public std::ranges::view_interface<view_take<urng_t, exactly,
     {
         return target_size;
     }
-}; // class view_take
```
Comment (MitraDarja) [inline include/seqan3/range/views/take.hpp:241]: Why deleted?

File: include/seqan3/range/views/take.hpp [hunk 7]
```diff
@@ -466,14 +470,7 @@ class view_take<urng_t,
         return base_base_t::operator[](n);
     }
     //!\}
-}; // class view_take::iterator_type
```
Comment (MitraDarja) [inline include/seqan3/range/views/take.hpp:469]: Why deleted?

File: include/seqan3/range/views/pairwise_combine.hpp (hunk 1 above)
Comment (marehr) [reply include/seqan3/range/views/pairwise_combine.hpp:72]: If you `default` constructors they implicitly are `constexpr` and `noexcept`, if they can be `constexpr` and `noexcept`.

Svenja requested this change in https://github.com/seqan/seqan3/pull/1914#discussion_r441996820 (the PR which override my changes) for the view::iterator class, I made this consistent across the file.

File: include/seqan3/range/views/repeat.hpp (hunk 3 above)
Comment (marehr) [reply include/seqan3/range/views/repeat.hpp:209]: It isn't in our style guide to do an end comment for classes. We do this only for namespaces and macros. I removed this for consistency.

**Comment by marehr** [comment]
> Plus, is the coverage error a problem? I always don't know, how to treat them as a Reviewer. :see_no_evil:

I looked at the report and didn't see any newly introduced misses in the change. This normally means that we removed code that had coverage before. If I had reviewed it I would dismiss this as a false positive.
